tests/core/filtering/test_contract_data_filters.py

```diff
@@ -162,37 +177,45 @@ def test_data_filters_with_list_arguments(
         emitter_event_ids,
         call_as_instance,
         create_filter,
+        api_style,
         vals):
 
     matching, non_matching = vals
 
     if call_as_instance:
-        event_filter = create_filter(emitter, [
-            'LogListArgs', {
-                'filter': {"arg1": matching}}])
+        contract = emitter
     else:
-        event_filter = create_filter(Emitter, [
-            'LogListArgs', {
-                'filter': {"arg1": matching}}])
-
-    txn_hashes = []
-    txn_hashes.append(emitter.functions.logListArgs(
-        arg0=matching,
-        arg1=matching).transact())
-    txn_hashes.append(emitter.functions.logListArgs(
-        arg0=non_matching,
-        arg1=non_matching).transact())
-    txn_hashes.append(emitter.functions.logListArgs(
-        arg0=non_matching,
-        arg1=matching).transact())
-    txn_hashes.append(emitter.functions.logListArgs(
-        arg0=matching,
-        arg1=non_matching).transact())
-
-    for txn_hash in txn_hashes:
-        wait_for_transaction(web3, txn_hash)
-
-    log_entries = event_filter.get_new_entries()
-    assert len(log_entries) == 2
-    assert log_entries[0]['transactionHash'] == txn_hashes[0]
-    assert log_entries[1]['transactionHash'] == txn_hashes[3]
+        contract = Emitter
+
+    if api_style == 'build_filter':
+        filter_builder = contract.events.LogListArgs.build_filter()
+        filter_builder.args['arg1'].match_single(matching)
+        event_filter = filter_builder.deploy(web3)
+
+        txn_hashes = []
+        txn_hashes.append(emitter.functions.logListArgs(
+            arg0=matching,
+            arg1=matching).transact())
+        txn_hashes.append(emitter.functions.logListArgs(
+            arg0=non_matching,
+            arg1=non_matching).transact())
+        txn_hashes.append(emitter.functions.logListArgs(
+            arg0=non_matching,
+            arg1=matching).transact())
+        txn_hashes.append(emitter.functions.logListArgs(
+            arg0=matching,
+            arg1=non_matching).transact())
+
+        for txn_hash in txn_hashes:
+            wait_for_transaction(web3, txn_hash)
+
+        log_entries = event_filter.get_new_entries()
+        assert len(log_entries) == 2
+        assert log_entries[0]['transactionHash'] == txn_hashes[0]
+        assert log_entries[1]['transactionHash'] == txn_hashes[2]
+    else:
+        with pytest.raises(TypeError):
```

Comment:
List arguments are forbidden for v3/v4 api's to prevent ambiguity between list arguments and lists match options.